src/java/org/apache/cassandra/io/sstable/format/SSTableFormat.java

```diff
@@ -113,99 +114,85 @@ IScrubber getScrubber(ColumnFamilyStore cfs,
 
     void deleteOrphanedComponents(Descriptor descriptor, Set<Component> components);
 
-    void setup(int id, String name, Map<String, String> options);
-
-    Type getType();
-
     /**
      * Deletes the existing components of the sstables represented by the provided descriptor.
      * The method is also responsible for cleaning up the in-memory resources occupied by the stuff related to that
      * sstables, such as row key cache entries.
      */
     void delete(Descriptor descriptor);
 
+    /**
+     * This class is not completely redundant
```

Comment:
I do feel that this is redundant, its just another reference to the values in `org.apache.cassandra.config.DatabaseDescriptor#sstableFormats` and 100% depends on that...

`current` -> `DD.getSelectedSSTableFormat`
`getByName` -> `DD.sstableFormats.get(name)`
`getByClass` -> 

```
DatabaseDescriptor.getSSTableFormats().values().stream()
                              .filter(f -> f.getClass().equals(formatClass))
                              .findFirst()
                              .orElseGet(() -> {
                                  throw new NoSuchElementException("Unknown sstable format class: " + formatClass);
                              });
```

Reply:
Did I really write "not"??? 🤦🏻‍♂️ I wanted to write it is redundant and that's why I marked it deprecated. That "not" is a typing mistake.

The only reason I haven't removed it is that the additional changeset would be quite large. So if we agree on that, we can do that.

Reply:
I am cool with you leaving this until the review is complete, then fixing at the end.